apps/events/viewsets.py

```diff
@@ -102,16 +102,29 @@ def get_template_names(self):
 
     def get_queryset(self):
         if self.kwargs.get("pk"):
-            self.queryset = Event.objects.filter(id=self.kwargs["pk"])
+            self.queryset = Event.objects.filter(
+                Q(id=self.kwargs["pk"]) & Q(is_visible=True) & Q(is_finished=False) & Q(type="open") |
+                Q(id=self.kwargs["pk"]) & Q(participants__in=[self.request.user.id]) & Q(is_finished=False) |
+                Q(id=self.kwargs["pk"]) & Q(created_by=self.request.user) & Q(type="private") |
+                Q(id=self.kwargs["pk"]) & Q(type="private") & Q(is_finished=True) & Q(is_visible=True)
```

Comment:
Maybe it should be like this:
1) For open events allow all users see them no matter if they are finished or not
2) Participant user can see events without conditions, event if they are finished, because he was part of it
3) Owner can see his events no matter what

Reply:
1. Done
2. 3. Allowed without conditions. "Private" conditions checks on permission layer